src/trivia/util.h

```diff
@@ -570,6 +570,41 @@ is_exp_of_two(unsigned n)
 	return (n & (n - 1)) == 0;
 }
 
+static inline char **
+str_split(char *str, const char *delimiter, unsigned *size, bool copy)
+{
+	char **tmp, **rc = NULL;
+	char *part = strtok((char *)str, delimiter);
+	*size = 0;
+
+	while (part != NULL) {
+		tmp = (char **)realloc(rc, ((++(*size)) * sizeof(char *)));
+		if (tmp == NULL)
+			goto error;
+
+		rc = tmp;
+		if (copy) {
+			rc[*size - 1] = strdup(part);
```

Comment:
A lot of reallocations and memory copying. I would consider this function as a candidate for refactoring (but maybe I am out of context so can't say for sure).